Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -1185,6 +1218,13 @@ void TextureMapperLayer::setBoundsOrigin(const FloatPoint& boundsOrigin)
 
 void TextureMapperLayer::setSize(const FloatSize& size)
 {
+#if ENABLE(DAMAGE_TRACKING)
+    if (m_state.size != size) {
+        // When layer size changes, we damage whole layer for now.
+        // FIXME: Damage only affected area.
+        m_damage.add(FloatRect(FloatPoint::zero(), size));
```

Comment:
Can m_damage be invalid here? Maybe we should do this only when tracking damaging?

Reply:
Good point.
Regarding doing this only when damage propagation is on - it will require coordination of implementation with the damage flow from `GraphicsLayerCoordinated` so I propose to do it in a separate change.